src/esockd_proxy_protocol.erl

```diff
@@ -106,27 +119,101 @@ parse_v2(?LOCAL, _Trans, _ProxyInfo, #proxy_socket{socket = Sock}) ->
 
 parse_v2(?PROXY, ?STREAM, ProxyInfo, ProxySock = #proxy_socket{inet = inet4}) ->
     <<A:8, B:8, C:8, D:8, W:8, X:8, Y:8, Z:8,
-      SrcPort:16, DstPort:16, _/binary>> = ProxyInfo,
-    {ok, ProxySock#proxy_socket{src_addr = {A, B, C, D}, src_port = SrcPort,
-                                dst_addr = {W, X, Y, Z}, dst_port = DstPort}};
+      SrcPort:16, DstPort:16, AdditionalBytes/binary>> = ProxyInfo,
+    parse_pp2_additional(AdditionalBytes, ProxySock#proxy_socket{
+        src_addr = {A, B, C, D}, src_port = SrcPort,
+        dst_addr = {W, X, Y, Z}, dst_port = DstPort});
 
 parse_v2(?PROXY, ?STREAM, ProxyInfo, ProxySock = #proxy_socket{inet = inet6}) ->
     <<A:16, B:16, C:16, D:16, E:16, F:16, G:16, H:16,
       R:16, S:16, T:16, U:16, V:16, W:16, X:16, Y:16,
-      SrcPort:16, DstPort:16, _/binary>> = ProxyInfo,
-    {ok, ProxySock#proxy_socket{src_addr = {A, B, C, D, E, F, G, H}, src_port = SrcPort,
-                                dst_addr = {R, S, T, U, V, W, X, Y}, dst_port = DstPort}};
+      SrcPort:16, DstPort:16, AdditionalBytes/binary>> = ProxyInfo,
+    parse_pp2_additional(AdditionalBytes, ProxySock#proxy_socket{
+        src_addr = {A, B, C, D, E, F, G, H}, src_port = SrcPort,
+        dst_addr = {R, S, T, U, V, W, X, Y}, dst_port = DstPort});
 
 parse_v2(_, _, _, #proxy_socket{socket = Sock}) ->
     esockd_transport:fast_close(Sock),
     {error, unsupported_proto_v2}.
 
+parse_pp2_additional(<<>>, ProxySock) ->
+    {ok, ProxySock};
+parse_pp2_additional(Bytes, ProxySock) when is_binary(Bytes) ->
+    IgnoreGuard = fun(?PP2_TYPE_NOOP) -> false; (_Type) -> true end,
+    AdditionalInfo = parse_pp2_tlv(fun pp2_additional_field/1, Bytes, IgnoreGuard),
+    {ok, ProxySock#proxy_socket{pp2_additional_info = AdditionalInfo}}.
+
+parse_pp2_tlv(Fun, Bytes) ->
+    parse_pp2_tlv(Fun, Bytes, fun(_Any) -> true end).
+parse_pp2_tlv(Fun, Bytes, Guard) ->
+    [Fun({Type, Val}) || <<Type:8, Len:16, Val:Len>> <= Bytes, Guard(Type)].
```

Comment:
the correct binary matching should be **<<Type:8, Len:16, Val:Len/binary>>** I think ?